### DISCUSSION TITLE
Undo/Redo actions stop working in IntelliJ IDEA when GitHub Copilot plug-in is enabled

### DISCUSSION THREAD

**Discussion by CrazyCoder** [community/community#49435] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Reported by many users at https://youtrack.jetbrains.com/issue/IDEA-312283/Cannot-undo-redo-when-using-GH-Copilot.

Nothing related in the logs.

Disabling plug-in fixed Undo/Redo issue in the IDE.

**Comment by theY4Kman** [comment]
FYI: I hacked around a fix for this by adding a try/catch to the offending `.class` file; I wrote about it in [this discussion](https://github.com/orgs/community/discussions/48885#discussioncomment-5249220)

**Reply by jpveilleux** [reply]
This really needs to be fixed. I appreciate the fix @theY4Kman did but we should not have to do this...

**Reply by theY4Kman** [reply]
fuckin agreed, especially since the fix is a grand total of one try/catch

**Comment by kodigilo** [comment]
I have the same problem

**Reply by CarsonF** [reply]
Just upvote the original post. It's frowned upon to comment saying "same" which spam notifications to everyone here. Upvoting is exactly for this purpose.

**Comment by somepeople** [comment]
Same thing. I've just update Webstorm to the last version and now, undo/redo doesn't work anymore. I've disabled the Copilot plugin and it works again.

**Comment by guiramos** [comment]
I am having the same issue.

**Comment by maksimu** [comment]
Yeah, I did remove the plugin, and that started to work.... but now I feel dumb 😃

**Comment by ericklombardo** [comment]
You could try this fix, it works for me 
https://youtrack.jetbrains.com/issue/IDEA-312283/Cannot-undo-redo-when-using-GitHub-Copilot-plugin#focus=Comments-27-7059632.0-0

**Comment by hstaudacher** [comment, marked as answer]
We have push GitHub Copilot 1.2.5.2507 to the JetBrains Marketplace. It's currently under review which might take a bit. The version contains a workaround for [IDEA-316656](https://youtrack.jetbrains.com/issue/IDEA-316656) which should solve this problem.

**Reply by guiramos** [reply]
Can you please let us when it's available?

**Reply by theY4Kman** [reply]
Excellent! Thank you! And thanks for creating the JB ticket, which I should've done once I realized the thrown `UninitializedPropertyAccessException` is out of plugins' control and definitely unintentional >_>

**Reply by dan-lee** [reply]
It's already released

**Reply by bcExpt1123** [reply]
Thank you!

**Comment by ghost** [comment]
I'm on plugin version 1.5.30-242 and still have the same problem.

**Reply by Broutard** [reply]
Same here with version 1.5.30-242
@hstaudacher

**Reply by tecnocat** [reply]
Same issue here

GitHub Copilot **1.5.31-242**

PhpStorm **2024.3.2**

Build #PS-243.23654.115, built on January 15, 2025

Licensed to **Redacted**
Subscription is active until April 22, 2025.

Runtime version: 21.0.5+8-b631.30 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

**Reply by Eskibear** [reply]
@tecnocat Can you try with the latest 1.5.32? Pls clear the cache if there’s any. I personally cannot reproduce it, but someone else mentioned after cleaning cache all work well.

**Reply by tecnocat** [reply]
Cleanning cache works.

Thanks!

**Reply by askoog** [reply]
Same issue here. Using 1.5.32-242 version of copilot plugin. Cleaning cache works for a short while, but the problem reappears after a while.

Seems like the undo/redo history becomes garbled somehow. Sometimes when i do "undo" the editor removes code in other places, removes a very old change and not the latest or refuses to undo anything at all

**Reply by jiec-msft** [reply]
@askoog I didn't reproduce this now. Could you also share your IDE information by "Help | About" and remove your personal information? I'll try reproduce with same env.

**Reply by Broutard** [reply]
> Same issue here. Using 1.5.32-242 version of copilot plugin. Cleaning cache works for a short while, but the problem reappears after a while.
> 
> Seems like the undo/redo history becomes garbled somehow. Sometimes when i do "undo" the editor removes code in other places, removes a very old change and not the latest or refuses to undo anything at all

I confirm, same here ! I cannot use my Copilot account since many weeks.

**Reply by jiec-msft** [reply]
@Broutard Thank you for confirmation. We pushed some fixes in `1.5.32-242` version of copilot, if you haven't upgraded to it, please have a try. And unfortunately, I cannot reproduce the undo/redo issue in my side. There can be possibility that it relates to the conflicts with the IDE's bundled plugins and other third-party plugins, could you also share your IDE information by copy paste "Help | About" from your IDE (kind reminder: please remove your personal information)?

**Reply by askoog** [reply]
> @askoog I didn't reproduce this now. Could you also share your IDE information by "Help | About" and remove your personal information? I'll try reproduce with same env.

```
IntelliJ IDEA 2024.3.2.2 (Ultimate Edition)
Build #IU-243.23654.189, built on January 29, 2025
Licensed to XXXXXXXXX
Subscription is active until XXXXXXX
Runtime version: 21.0.5+8-b631.30 aarch64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.lwawt.macosx.LWCToolkit
macOS 15.3
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 6144M
Cores: 10
Metal Rendering is ON
Registry:
  undo.documentUndoLimit=500
  debugger.new.tool.window.layout=true
  ide.experimental.ui=true
  terminal.new.ui=true
Non-Bundled Plugins:
  com.jetbrains.space (243.23654.19)
  com.intellij.lang.puppet (243.21565.122)
  Builder Generator (1.4.0)
  Key Promoter X (2024.2.2)
  net.sf.opk.avro-schema-support (232.0.2)
  com.intellij.ml.llm (243.23654.269.5)
  com.yiycf.plugins.mavenDependencyHelper (2.5)
  MavenRunHelper (4.29.0-IJ2022.2)
  com.github.copilot (1.5.32-242)
Kotlin: 243.23654.189-IJ
```

Hmmm, I see there's an entry for increasing `documentUndoLimit`. I don't recall changing that one.

**Reply by Broutard** [reply]
> @Broutard Thank you for confirmation. We pushed some fixes in `1.5.32-242` version of copilot, if you haven't upgraded to it, please have a try. And unfortunately, I cannot reproduce the undo/redo issue in my side. There can be possibility that it relates to the conflicts with the IDE's bundled plugins and other third-party plugins, could you also share your IDE information by copy paste "Help | About" from your IDE (kind reminder: please remove your personal information)?

PhpStorm 2024.3.2.1
Build #PS-243.23654.168, built on January 24, 2025
Licensed to XXXXXXXXX
Subscription is active until XXXXXXXXX
Runtime version: 21.0.5+8-b631.30 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.X11.XToolkit
Linux 6.8.0-52-generic
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 8192M
Cores: 16
Registry:
  ide.balloon.shadow.size=0
  editor.focus.mode.color.light=415967
  editor.focus.mode.color.dark=415967
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.new.editor.tabs.vertical.borders=true
  ide.experimental.ui=true
  ide.editor.tab.selection.animation=true
  ide.images.show.chessboard=true
Non-Bundled Plugins:
  de.espend.idea.localization (0.0.6)
  com.nasller.CodeGlancePro (1.9.6)
  String Manipulation (9.15.0)
  com.intellij.ml.llm (243.23654.269.5)
  izhangzhihao.rainbow.brackets (2024.2.8-241)
  com.chrisrm.idea.MaterialThemeUI (9.7.0)
  ru.adelf.idea.dotenv (2024.3)
  de.espend.idea.php.annotation (11.1.1)
  fr.adrienbrault.idea.symfony2plugin (2024.1.276)
  com.github.copilot (1.5.32-242)
  com.mallowigi (101.2.0)
  mobi.hsz.idea.gitignore (4.5.4)
Current Desktop: ubuntu:GNOME

**Reply by jiec-msft** [reply]
@askoog @Broutard Unfortunately we still cannot reproduce it, but we'll keep an eye on it when we dogfood our products.

**Comment by thakkarnetram** [comment]
Same issue in 2024.3.1.1

**Comment by OvermindDL1** [comment]
Same issue in 2025.1.5.  Disabling the Github Copilot extension then undo/redo works again, re-enabling the Github Copilot extension causes the issue again.  Have performed a new installation of the Jetbrains Rustrover IDE with fresh settings and all and problem still occasionally happens.

**Comment by vultix** [comment]
I'm seeing the same issue as well. @hstaudacher Should we open a new issue?

**Comment by combea** [comment]
I have the issue again, in plugin v1.5.57-243 and IntelliJ IDEA 2025.2.2 Ultimate. It does not happen on every projects I have, but when it does it's really annoying
